### PR TITLE
pythonPackages.ledgerwallet: init at 0.1.2

### PR DESCRIPTION
**Pull request by d-xo**
###### Motivation for this change

adds the [ledgerwallet](https://pypi.org/project/ledgerwallet) python package.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ledgerwallet/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    owner = "LedgerHQ";
+    repo = "ledgerctl";
+    rev = "v${version}";
+    sha256 = "0fb93h2wxm9as9rsywlgz2ng4wrlbjphn6mgbhj6nls2i86rrdxk";
+  };
+
+  pythonImportsCheck = [ "ledgerwallet" ];
+
+  buildInputs = [ libusb ];
+  propagatedBuildInputs = [ cryptography click construct ecdsa hidapi intelhex pillow protobuf requests tabulate ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ledgerwallet/default.nix:31]:
```suggestion
  buildInputs = [ libusb ];
  propagatedBuildInputs = [ cryptography click construct ecdsa hidapi intelhex pillow protobuf requests tabulate ];

  pythonImportsCheck = [ "ledgerwallet" ];
```

Just a nitpick.

File: pkgs/development/python-modules/ledgerwallet/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  meta = {
+    homepage = "https://github.com/LedgerHQ/ledgerctl";
+    description = "A library to control Ledger devices";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ xwvvvvwx ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ledgerwallet/default.nix:37]:
```suggestion
  meta = with lib; {
    homepage = "https://github.com/LedgerHQ/ledgerctl";
    description = "A library to control Ledger devices";
    license = licenses.mit;
    maintainers = with maintainers; [ xwvvvvwx ];
```

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
I needed to fetch [`0cc95f6` (#102306)](https://github.com/NixOS/nixpkgs/pull/102306/commits/0cc95f6b1937383118507c6ae549d210f96cea75) and https://github.com/NixOS/nixpkgs/pull/104565 to be able to build it on darwin.

Also can you add a ``pythonImportsCheck``? The package does not ship any tests so we make sure it can at least be imported.

File: pkgs/development/python-modules/ledgerwallet/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    repo = "ledgerctl";
+    rev = "v${version}";
+    sha256 = "0fb93h2wxm9as9rsywlgz2ng4wrlbjphn6mgbhj6nls2i86rrdxk";
+  };
+
+  propagatedBuildInputs = [ cryptography click construct ecdsa hidapi intelhex pillow protobuf requests tabulate ];
+
+  pythonImportsCheck = [ "ledgerwallet" ];
+
+  meta = with lib; {
```
[the remark this answers is not in the document]
Comment (xwvvvvwx) [reply pkgs/development/python-modules/ledgerwallet/default.nix:31]: Is there any semantic difference?

**Comment by xwvvvvwx** [comment]
oh nice! I don't have a mac to test on, much appreciated :)

File: pkgs/development/python-modules/ledgerwallet/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/ledgerwallet/default.nix:31]: For darwin I needed stdenv and I did not want to pass in stdenv and lib.

**Comment by xwvvvvwx** [comment]
review comments addressed.

> Also can you add a pythonImportsCheck? The package does not ship any tests so we make sure it can at least be imported.

This was already there I think, unless I'm not understanding what you meant?

**Comment by xwvvvvwx** [comment]
Is there anything still pending here?

**Comment by xwvvvvwx** [comment]
Bumping, is there anything more to do here?

**Comment by markuskowa** [comment]
@SuperSandro2000 is this ready to merge?

**Comment by markuskowa** [comment]
@xwvvvvwx There's still an evaluation error. A rebase may help?

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -3355,6 +3355,8 @@ in {
 
   ledgerblue = callPackage ../development/python-modules/ledgerblue { };
 
+  ledgerwallet = callPackage ../development/python-modules/ledgerwallet { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3358]:
```suggestion
  ledgerwallet = callPackage ../development/python-modules/ledgerwallet {
    inherit (pkgs.darwin.apple_sdk.frameworks) AppKit;
  };
```

File: pkgs/development/python-modules/ledgerwallet/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "LedgerHQ";
+    repo = "ledgerctl";
+    rev = "v${version}";
+    sha256 = "0fb93h2wxm9as9rsywlgz2ng4wrlbjphn6mgbhj6nls2i86rrdxk";
+  };
+
+  buildInputs = stdenv.lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ AppKit ]);
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ledgerwallet/default.nix:28]:
```suggestion
  buildInputs = stdenv.lib.optionals stdenv.isDarwin [ AppKit ];
```

File: pkgs/development/python-modules/ledgerwallet/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+, click
+, construct
+, ecdsa
+, hidapi
+, intelhex
+, pillow
+, protobuf
+, requests
+, tabulate
+, darwin
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ledgerwallet/default.nix:14]:
```suggestion
, AppKit
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104506` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python39Packages.ledgerwallet</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.ledgerwallet</li>
    <li>python38Packages.ledgerwallet</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104506` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python39Packages.ledgerwallet</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.ledgerwallet</li>
    <li>python38Packages.ledgerwallet</li>
  </ul>
</details>

**Comment by xwvvvvwx** [comment]
Thanks @SuperSandro2000 :tada:

**Pull request merged**